test/e2e/e2e_test.go

```diff
@@ -34,3 +40,26 @@ var _ = BeforeSuite(func() {
 		panic(err)
 	}
 })
+
+var _ = AfterSuite(func() {
+	// Only attempt deletion of HCP cluster if a cluster was actually created
+	if e2eSetup.Cluster.Name == "" || e2eSetup.CustomerEnv.CustomerRGName == "" {
+		return
+	}
+	log.Logger.Infof("Starting deletion of cluster %s in resource group %s...", e2eSetup.Cluster.Name, e2eSetup.CustomerEnv.CustomerRGName)
+	ctxDel, cancel := context.WithTimeout(context.Background(), 45*time.Minute)
+	defer cancel()
+	hcpClient := clients.NewHcpOpenShiftClustersClient()
+	clusterName := e2eSetup.Cluster.Name
+	resourceGroup := e2eSetup.CustomerEnv.CustomerRGName
+	poller, err := hcpClient.BeginDelete(ctxDel, resourceGroup, clusterName, nil)
+	if err != nil {
+		panic(fmt.Sprintf("Cluster deletion should succeed (begin): %v", err))
+	}
+	_, err = poller.PollUntilDone(ctxDel, &runtime.PollUntilDoneOptions{
+		Frequency: 10 * time.Second,
+	})
+	if err != nil {
+		panic(fmt.Sprintf("Cluster deletion should succeed (poll): %v", err))
+	}
+})
```

Comment:
if you wire creation through a helper, then deletion can be auto-registered with any number of resource groups.  See example here.

https://github.com/Azure/ARO-HCP/blob/main/test/util/framework/per_test_framework.go#L126  with usage here https://github.com/Azure/ARO-HCP/blob/main/test/e2e/complete_cluster_create.go#L52 

```go
			resourceGroup, err := ic.NewResourceGroup(ctx, "basic-create", "uksouth")
			Expect(err).NotTo(HaveOccurred())
```

That block creates a non-conflicting resource group and will auto-clean up when the test is over.